app/ts/store/account.ts

```diff
@@ -57,6 +58,20 @@ async function connect() {
 	}
 }
 
+async function ensureConnected() {
+ 	try {
+ 		assertsExternalProvider(window.ethereum)
+ 		const provider = new ethers.providers.Web3Provider(window.ethereum)
+ 		const signer = provider.getSigner()
+ 		store.value = { status: 'busy' }
+ 		const address = await signer.getAddress()
+ 		store.value = { status: 'connected', address }
+ 	} catch (exception) {
+ 		store.value = { status: 'disconnected', connect, ensureConnected }
+ 	}
```

Comment:
Consider using `useAsyncState` rather than manually reconstructing the logic everywhere that you have some async work.  Same for `connect()` above.

Reply:
Refactoring this seem to have some moving parts that needed to be tested outside of this changeset. created a separate issue #54